include/ProgramRepresentation/PVAliasSet.h

```diff
@@ -0,0 +1,71 @@
+#ifndef PV_ALIAS_SET_H
+#define PV_ALIAS_SET_H
+
+#include "ProgramRepresentation/MethodsSet.h"
+#include "ProgramRepresentation/ProgramVariable.h"
+#include "Utils.h"
+
+// holds a set of program variables and methods called on them.
+// each program variable in the set is considered an alias set.
+class PVAliasSet {
+private:
+  std::list<ProgramVariable> programVariables;
+
+  // holds the methods called on the set
+  MethodsSet methods;
+
+public:
+  PVAliasSet();
+
+  // returns true iff programVar is in the set by checking its raw name
+  bool contains(ProgramVariable programVar);
+
+  // returns true iff there exists a program variable in our set with raw name
+  // equal to rawName
+  bool containsByRawName(const std::string &rawName);
+
+  // returns true iff there exists a program variable in our set with clean name
+  // equal to cleanedName
+  bool containsByCleanedName(const std::string &cleanedName);
+
+  // adds programVar to our set of program variables. programVar is checked to
+  // see if it already exists in our set of program variables and it's not added
+  // if it already does.
+  void add(ProgramVariable programVar);
+
+  // adds array of program variables to our set of program variables.
+  // each variable is checked to see if already exists in our list of program
+  // variables.
+  void addProgramVariables(std::list<ProgramVariable> programVariables);
+
+  std::list<ProgramVariable> getProgramVariables();
+
+  MethodsSet getMethodsSet() const;
+  MethodsSet *getMethodsSetRef();
+
+  void setMethodsSet(MethodsSet methods);
+
+  // generates the set of program variables represented by their cleaned name as
```

Comment:
sorry, I should have removed this comment. The name is fine as-is, although personally I would've left off "generate" (since I don't think it adds much to the name) and just called this `toString` or similar.